nrf_rpc/include/nrf_rpc.h

```diff
@@ -0,0 +1,444 @@
+struct nrf_rpc_tr_remote_ep *_nrf_rpc_evt_prep(
+	const struct nrf_rpc_group *group);
+void _nrf_rpc_evt_alloc_error(struct nrf_rpc_tr_remote_ep *tr_remote_ep);
+void _nrf_rpc_evt_unprep(struct nrf_rpc_tr_remote_ep *tr_remote_ep);
+struct nrf_rpc_tr_remote_ep *_nrf_rpc_rsp_prep();
+
+
+/** @brief Initialize the nRF RPC
+ *
+ * @returns         NRF_RPC_SUCCESS or error code from enum nrf_rpc_error_code.
+ */
+int nrf_rpc_init(void);
+
+
+/** @brief Send a command.
+ *
+ * @param remote_ep    Destination endpoint allocated by @a NRF_RPC_CMD_ALLOC.
+ * @param cmd          Command id.
+ * @param packet       Packet allocated by @a NRF_RPC_CMD_ALLOC and filled with
+ *                     encoded data.
+ * @param len          Length of the packet. Can be smaller than allocated.
+ * @param handler      Callback that handles the response. In case of error
+ *                     it is undefined if the handler will be called.
+ * @param handler_data Opaque pointer that will be passed to @a handler.
+ * @returns            NRF_RPC_SUCCESS or error code from
+ *                     enum nrf_rpc_error_code.
+ */
+int nrf_rpc_cmd_send(nrf_rpc_alloc_ctx ctx, uint8_t cmd, uint8_t *packet,
+		     size_t len, nrf_rpc_handler_t handler, void *handler_data);
+
+
+/** @brief Send a command and get response directly.
+ *
+ * After successful return caller is resposible for decoding content of the
+ * response packet an call @a nrf_rpc_decoding_done just after that. After
+ * calling @a nrf_rpc_decoding_done @a rsp_packet is no longer valid.
+ *
+ * Depending on transport layer implementation this function may be slightly
+ * slower than @a nrf_rpc_cmd_send, because additional thread context switching
+ * may happen.
+ *
+ * @param remote_ep       Destination endpoint allocated @a NRF_RPC_CMD_ALLOC.
+ * @param cmd             Command id.
+ * @param packet          Packet allocated by @a NRF_RPC_CMD_ALLOC and filled
+ *                        with encoded data.
+ * @param len             Length of the packet. Can be smaller than allocated.
+ * @param[out] rsp_packet Response packet.
+ * @param[out] rsp_len    Response packet length.
+ * @returns               NRF_RPC_SUCCESS or error code from
+ *                        enum nrf_rpc_error_code.
+ */
+int nrf_rpc_cmd_rsp_send(nrf_rpc_alloc_ctx ctx, uint8_t cmd, uint8_t *packet,
+			 size_t len, const uint8_t **rsp_packet,
+			 size_t *rsp_len);
+
+
+/** @brief Send a command passing any errors to an error handler.
+ *
+ * If error occurred during sending this command it will be passed to
+ * @a nrf_rpc_error_handler instead of returning it. This form of error handling
+ * can be useful for serializing API function that has no ability to report
+ * an error in any other way, e.g. it returns void.
+ *
+ * See @a nrf_rpc_cmd_send for more details.
+ */
+void nrf_rpc_cmd_send_noerr(nrf_rpc_alloc_ctx ctx, uint8_t cmd, uint8_t *packet,
+			    size_t len, nrf_rpc_handler_t handler,
+			    void *handler_data);
```

Comment:
Yes, user can handle it in own error handling system, but there is also nRF RPC error handling system, so this variant of the function is a shortcut for the user if he wants to use ours error handling system.

I see a use case: It will be often used in BLE API serialization in functions that returns void.

Reply:
It can be done this way, but I think that repeating the same pattern in many functions should be avoided if possible. `nrf_rpc_cmd_send_noerr` function is a util to avoid it. That is why I prefer to keep it.

Reply:
In the newest updates I keep the `_noerr` variants. I think that if the user wants to pass errors generated in nRF RPC to nRF RPC error handling system without additional code then he should have this option. Can you @trond-snekvik agree to keep those functions?